src/components/base/b-virtual-scroll/test/unit/functional/rendering/items-factory.ts

```diff
@@ -0,0 +1,299 @@
+					}
+				}));
+			});
+
+			await component
+				.withDefaultPaginationProviderProps({chunkSize})
+				.withProps({
+					itemsFactory,
+					shouldPerformDataRender: () => true,
+					chunkSize
+				})
+				.build();
+
+			await component.waitForContainerChildCountEqualsTo(chunkSize);
+
+			await test.expect(component.childList).toHaveCount(chunkSize);
+		});
+	});
+
+	test.describe('In additional `item`, `separator` was also returned', () => {
+		test('Should render both', async () => {
+			const
+				chunkSize = 12;
+
+			provider
+				.responseOnce(200, {data: state.data.addData(chunkSize)})
+				.response(200, {data: state.data.addData(0)});
+
+			const separator = {
+				item: 'b-button',
+				key: '',
+				children: {
+					default: 'ima button'
+				},
+				props: {
+					id: 'button'
+				},
+				type: 'separator'
+			};
+
+			const itemsFactory = await component.mockFn((state, ctx, separator) => {
+				const
+					data = state.lastLoadedData;
+
+				const items = data.map((item) => ({
+					item: 'section',
+					key: Object.cast(undefined),
+					type: 'item',
+					children: [],
+					props: {
+						'data-index': item.i
+					}
+				}));
+
+				items.push(separator);
+
+				return items;
+			}, separator);
+
+			await component
+				.withDefaultPaginationProviderProps({chunkSize})
+				.withProps({
+					itemsFactory,
+					shouldPerformDataRender: () => true,
+					chunkSize
+				})
+				.build();
+
+			await component.waitForContainerChildCountEqualsTo(chunkSize + 1);
+
+			await test.expect(component.container.locator('#button')).toBeVisible();
+			await test.expect(component.childList).toHaveCount(chunkSize + 1);
+		});
+	});
+
+	test.describe('Returned items with type `item` is less than the provided data', () => {
+		test('Should render items that was returned from `itemsFactory`', async () => {
+			const
+				chunkSize = 12,
+				renderedChunkSize = chunkSize - 2;
+
+			provider
+				.responseOnce(200, {data: state.data.addData(chunkSize)})
+				.response(200, {data: state.data.addData(0)});
+
+			const itemsFactory = await component.mockFn<ComponentItemFactory<{i: number}>>((state) => {
+				const data = state.lastLoadedData;
+
+				const items = data.map<ComponentItem>((item) => ({
+					item: 'section',
+					key: Object.cast(undefined),
+					type: 'item',
+					children: [],
+					props: {
+						'data-index': item.i
+					}
+				}));
+
+				items.length -= 2;
+				return items;
+			});
+
+			await component
+				.withDefaultPaginationProviderProps({chunkSize})
+				.withProps({
+					itemsFactory,
+					shouldPerformDataRender: () => true,
+					chunkSize
+				})
+				.build();
+
+			await component.waitForContainerChildCountEqualsTo(renderedChunkSize);
+
+			await test.expect(component.childList).toHaveCount(renderedChunkSize);
+		});
+	});
+
+	test.describe('Returned item with type `item` is more than the provided data', () => {
+		test('Should render items that was returned from `itemsFactory`', async () => {
```

Comment:
```suggestion
		test('Should render items that were returned from `itemsFactory`', async () => {
```

Множественное число требует were